tests/prefer-vi-mocked.test.ts

```diff
@@ -0,0 +1,143 @@
+import rule, { RULE_NAME } from "../src/rules/prefer-vi-mocked";
+import { ruleTester } from "./ruleTester";
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    "foo();",
+    "vi.mocked(foo).mockReturnValue(1);",
+    "bar.mockReturnValue(1);",
+    "sinon.stub(foo).returns(1);",
+    "foo.mockImplementation(() => 1);",
+    "obj.foo();",
+    "mockFn.mockReturnValue(1);",
+    "arr[0]();",
+    "obj.foo.mockReturnValue(1);",
+    'vi.spyOn(obj, "foo").mockReturnValue(1);',
+    "(foo as Mock.vi).mockReturnValue(1);",
+    `type MockType = Mock;
+const mockFn = vi.fn();
+(mockFn as MockType).mockReturnValue(1);`,
+  ],
+  invalid: [
+    {
+      code: "(foo as Mock).mockReturnValue(1);",
```

Comment:
These are almost exactly the same as the test cases as in https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/__tests__/prefer-jest-mocked.test.ts
But with
```diff
-code: "(foo as jest.Mock).mockReturnValue(1);",
+code: "(foo as Mock).mockReturnValue(1);",
```